exam-1-study-guide.md

```diff
@@ -0,0 +1,250 @@
+null. Primitives start with lowercase while Reference types start with uppercase.
+
+**What are examples of reference types in Java?**
+
+String, Arrays, Classes, Interface, etc.
+
+**What is the first index in an array?**
+
+0
+
+**What are some properties of arrays?**
+
+- Size of arrays can’t be changed once declared.
+- Arrays can contain arrays
+- Arrays can store primitive types or objects
+- Easy and fast to access every element in an array
+
+**How can we declare an array literal?**
+
+```java
+int[]  arr = {0, 1, 2, 3, 4};
+```
+
+**How can we iterate over an array?**
+
+```java
+for (double element: values) {…}
+```
+
+**What is an operator? What mathematical functions can be used in Java?**
+
+An operator is symbol (or combination of a couple symbols) that is used with 
+variables and value to simplify how you write certain program expressions.
+
+**What is an expression? Where do they appear?**
+
+A “syntactically well formed and meaningful fragment” (roughly analogous to a 
+word in natural language. Examples include the following:
+
+- i
+- j+7
+- “Hello” + “World!”
+- keyboardIn.nextLine()		
+- n==0;
+- new SimpleWriter1L()
+
+**What is a statement?**
+
+A “smallest complete unit of execution” — which can be a complete unit of 
+manipulation, terminated by semicolon ‘ ; ’.
+
+**What is the difference between an expression and a statement?**
+
+A statement can contain expressions. The expression is some procedure of 
+calculation but a statement is a complete unit in a program. 
+
+**How can statements vary from each other?**
+
+TBD
+
+**What is the appropriate way to compare strings?**
+
+Best Practice: do not use == or != with strings. Use “equals” method.
+
+**What is the best way to compare doubles?**
+
+Best Practice: do not check doubles for equality
+
+**What are the most common operators?**
+
+| string | boolean | char            | int             | double |
+| ------ | ------- | --------------- | --------------- | ------ |
+|        | !       |                 | ++ --           |        |
+| +      | \|\|    |                 | + -             | + -    |
+|        | &&      |                 | * / %           | * /    |
+|        | == !=   | < > <= >= == != | < > <= >= == != | < >    |
+
+**What is control flow?**
+
+Change the flow of execution by adding conditional statements such as 
+if/if-else/switch and/or loops.
+
+**What are if statements?**
+
+Executes certain sections of code only if the particular test evaluates to 
+true or false.
+
+**What is a while loop?**
+
+Executes a body of code for as long as the condition statement is true.
+
+**What is a for loop?**
+
+A loop used when a body of code needs to be executed for a known number of times.
+
+**What is object-oriented programming?**
+
+A programming which is based on “objects”, like a combination of data and code.
+
+**What does the ‘static’ modifier do in a static method?**
+
+TBD
+
+**Where should you use a static method?**
+
+TBD
```

Comment:
A static method should be used where object information does not matter. If a method will always give the same outputs for a given input, it should be a static method.